client/src/main/kotlin/io/hstream/impl/ConsumerKtImpl.kt

```diff
@@ -241,7 +241,14 @@ class ConsumerKtImpl(
             try {
                 logger.info("consumer [{}] is starting", consumerName)
                 notifyStarted()
-                watchFuture = futureForIO { watchSubscription() }
+                watchFuture = (futureForIO { watchSubscription() }).handle { ret, err ->
+                    run {
```

Comment:
Why do you use the ``run`` function? is that necessary?

Reply:
I am not sure [18ee884](https://github.com/hstreamdb/hstreamdb-java/pull/82/commits/18ee884214d1c269214d15b1938c4b24959ce675)